src/array/cpu/spmm_binary_ops.h

```diff
@@ -148,7 +148,9 @@ constexpr DType Min<DType>::zero;
 
 #define SWITCH_BITS(bits, DType, ...)                           \
   do {                                                          \
-    if ((bits) == 16 || (bits) == 32) {                         \
+    if ((bits) == 16) {                                         \
+      LOG(FATAL) << "FP16 not supported on CPU";                \
```

Comment:
I'm afraid it will fail compilation if you do not define DType here.

Reply:
It won't because we don't have `__VA_ARGS__` in this branch.

Reply:
okay it's fine.